test/_generate_expected_node.js

```diff
@@ -1,30 +1,85 @@
-const canvg = require("../src/canvg.js"),
-  Canvas = require("canvas"),
-  path = require("path"),
-  Promise = require("bluebird"),
-  fs = Promise.promisifyAll(require("fs")),
-  PNGImage = Promise.promisifyAll(require("pngjs-image"));
+const canvg = require('../src/canvg.js'),
+  Canvas = require('canvas'),
+  path = require('path'),
+  Promise = require('bluebird'),
+  fs = Promise.promisifyAll(require('fs')),
+  PNGImage = Promise.promisifyAll(require('pngjs-image'));
 
-const file = process.argv[2];
-const fileName = path.resolve(`${__dirname}/../svgs/${file}`);
-if (!fs.existsSync(fileName)) {
-  throw new Error(`${fileName} does not exist!`);
+function createCanvas (width, height) { // 2.0 API
+  return new Canvas(width, height);
 }
 
+const file = process.argv[2];
+const start = process.argv[3] || 0;
+const end = process.argv[4];
+
+const basePath = `${__dirname}/../svgs/`;
+const files = file === 'all' ? fs.readdirSync(basePath) : [file];
+
+const fileNames = files.map((file) => {
+  return path.resolve(`${basePath}${file}`);
+});
+
 (async () => {
-  const svgbuffer = await fs.readFileAsync(fileName),
-    svg = svgbuffer.toString("utf-8"),
-    canvas = new Canvas(800, 600);
-
-  canvg(canvas, svg, {
-    ignoreMouse: true,
-    ignoreAnimation: true,
-    xmldom: {
-      errorHandler: function(/* level, msg */) {} // supress xmldom warnings
+  async function loadImageBuffers (svgbuffers) {
+    let images;
+    try {
+      console.log('Loading image buffers...');
+      images = await Promise.all(svgbuffers.map(([fileName, svgbuffer]) => {
```

Comment:
your fault error might come from the fact that you're storing all the images in memory and then writing them all to the file system. perhaps it's better to just gen the image and write it right away and try doing this in parallel.  also, with `Promise.all` you can't control how many run at once.  Maybe it's best to batch them in groups of 5 and this will start working.

Reply:
One can use the start/end parameters to do that, and as mentioned in a comment just now, the segfaults are occurring regardless of this PR...